### PR TITLE
Added test for _get_dtype_type.

### PR DESCRIPTION
**Pull request by topper-123**
- [ ] closes #xxxx
 - [ x] tests added / passed
 - [ x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff`` (On Windows, ``git diff upstream/master -u -- "*.py" | flake8 --diff`` might work as an alternative.)
 - [ ] whatsnew entry

Tests for ``core.dtype.common._get_dtype_type``.

### CODE REVIEW

File: pandas/tests/dtypes/test_common.py [hunk 1]
```diff
@@ -563,3 +563,37 @@ def test__get_dtype(input_param, result):
...
+    (pd.Series(['a', 'b']), np.object_),
+    (pd.Index([1, 2], dtype='int64'), np.int64),
+    (pd.Index(['a', 'b']), np.object_),
+    ('category', com.CategoricalDtypeType),
+    (pd.Categorical(['a', 'b']).dtype, com.CategoricalDtypeType),
+    (pd.Categorical(['a', 'b']), com.CategoricalDtypeType),
+    (pd.CategoricalIndex(['a', 'b']).dtype, com.CategoricalDtypeType),
+    (pd.CategoricalIndex(['a', 'b']), com.CategoricalDtypeType),
+    (pd.DatetimeIndex([1, 2]), np.datetime64),
+    (pd.DatetimeIndex([1, 2]).dtype, np.datetime64),
```
Comment (jreback) [inline pandas/tests/dtypes/test_common.py]: add a DTI with a tz

File: pandas/tests/dtypes/test_common.py [hunk 2]
```diff
@@ -563,3 +563,37 @@ def test__get_dtype(input_param, result):
...
+    ('datetime64[ns, Europe/London]', com.DatetimeTZDtypeType),
+    (pd.SparseSeries([1, 2], dtype='int32'), np.int32),
+    (pd.SparseSeries([1, 2], dtype='int32').dtype, np.int32),
+    (PeriodDtype(freq='D'), com.PeriodDtypeType),
+    ('period[D]', com.PeriodDtypeType),
+    (IntervalDtype(), com.IntervalDtypeType),
+    (None, type(None)),
+    (1, type(None)),
+    (1.2, type(None)),
+    (pd.DataFrame([1, 2]), type(None)),
```
Comment (jreback) [inline pandas/tests/dtypes/test_common.py]: put a comment on the last one as its a composite dtype

Comment (topper-123) [inline pandas/tests/dtypes/test_common.py]: Not sure I understand this.  just ``# composite dtype``?

**Comment by jreback** [comment]
meaning DataFrame doesn't have a dtype property

**Comment by jreback** [comment]
can you rebase and update

**Comment by topper-123** [comment]
Ok, I think it's alright now?

**Comment by jreback** [comment]
thanks @topper-123 nice tests!

**Pull request merged**